docs/ref/forms/api.txt

```diff
@@ -830,107 +830,211 @@ The field-specific output honors the form object's ``auto_id`` setting::
 
 .. attribute:: BoundField.id_for_label
 
-Use this property to render the ID of this field. For example, if you are
-manually constructing a ``<label>`` in your template (despite the fact that
-:meth:`~BoundField.label_tag` will do this for you):
+    Use this property to render the ID of this field. For example, if you are
+    manually constructing a ``<label>`` in your template (despite the fact that
+    :meth:`~BoundField.label_tag` will do this for you):
+
+    .. code-block:: html+django
+
+        <label for="{{ form.my_field.id_for_label }}">...</label>{{ my_field }}
+
+    By default, this will be the field's name prefixed by ``id_``
+    ("``id_my_field``" for the example above). You may modify the ID by setting
+    :attr:`~django.forms.Widget.attrs` on the field's widget. For example,
+    declaring a field like this::
+
+        my_field = forms.CharField(widget=forms.TextInput(attrs={'id': 'myFIELD'}))
+
+    and using the template above, would render something like:
+
+    .. code-block:: html
+
+        <label for="myFIELD">...</label><input id="myFIELD" type="text" name="my_field" />
+
+.. attribute:: BoundField.is_hidden
+
+    Returns ``True`` if this :class:`~django.forms.BoundField`'s widget is
+    hidden.
+
+.. attribute:: BoundField.label
+
+    The :attr:`~django.forms.Field.label` of the field. This is used in
+    :meth:`~BoundField.label_tag`.
+
+.. attribute:: BoundField.name
 
-.. code-block:: html+django
+    The name of this field in the form::
 
-    <label for="{{ form.my_field.id_for_label }}">...</label>{{ my_field }}
+        >>> f = ContactForm()
+        >>> print(f['subject'].name)
+        subject
+        >>> print(f['message'].name)
+        message
 
-By default, this will be the field's name prefixed by ``id_``
-("``id_my_field``" for the example above). You may modify the ID by setting
-:attr:`~django.forms.Widget.attrs` on the field's widget. For example,
-declaring a field like this::
+Methods of ``BoundField``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. method:: BoundField.as_hidden(attrs=None, **kwargs)
+
+    Returns a string of HTML for representing this as an ``<input type="hidden">``.
+
+    ``**kwargs`` are passed to :meth:`~django.forms.BoundField.as_widget`.
+
+.. method:: BoundField.as_text(attrs=None, **kwargs)
+
+    Returns a string of HTML for representing this as an ``<input type="text">``.
+
+    ``**kwargs`` are passed to :meth:`~django.forms.BoundField.as_widget`.
+
+.. method:: BoundField.as_textarea(attrs=None, **kwargs)
+
+    Returns a string of HTML for representing this as a ``<textarea>``.
+
+    ``**kwargs`` are passed to :meth:`~django.forms.BoundField.as_widget`.
+
+.. method:: BoundField.as_widget(widget=None, attrs=None, **kwargs)
+
+    Renders the field by rendering the passed widget, adding any HTML
+    attributes passed as ``attrs``.  If no widget is specified, then the
+    field's default widget will be used. ``**kwargs`` are only used by
```

Comment:
I wasn't sure if only_initial is used by django somewhere so I left the note there. 
It is aimed at developers of custom bound fields so that they pass the kwargs to the super method.